### PR TITLE
Template create CPU

### PR DESCRIPTION
**Pull request by janosdebugs**
## Please describe the change you are making

This PR adds CPU handling to templates in live and mock. It also adds correct handling for VMs when a template is used.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

File: newmock.go [hunk 1]
```diff
@@ -17,11 +17,53 @@ func NewMock() MockClient {
...
+	testStorageDomain.client = client
+	secondaryStorageDomain.client = client
+	testDatacenter.client = client
+	testNetwork.client = client
+	testVNICProfile.client = client
+
+	return client
+}
+
+func getClient(
```
Comment (eslutsky) [inline newmock.go:57]: What's the purpose of this getClient method? do we plan to use it somewhere else?

Comment (janosdebugs) [reply newmock.go:57]: It's refactoring the NewMock function so it isn't extremely long.

Comment (janosdebugs) [reply newmock.go:57]: I split it out of NewMock, so it doesn't become extremely long.

File: mock_template_create.go [hunk 2]
```diff
@@ -31,6 +32,7 @@ func (m *mockClient) CreateTemplate(
 		name:        name,
 		description: description,
 		status:      TemplateStatusLocked,
+		cpu:         vm.cpu.clone(),
```
Comment (eslutsky) [inline mock_template_create.go:35]: it appears that you cannot have several templates with the same name. we might  want to implement this constraint here and return an error in case the same name exists .. (we can put it on different PR)

**Review by eslutsky — APPROVED** [review]
looks good to me
